micrometer-core/src/test/java/io/micrometer/core/instrument/MeterFilterConfigPropertiesTest.java

```diff
@@ -0,0 +1,57 @@
+package io.micrometer.core.instrument;
+
+
+import io.micrometer.core.instrument.histogram.HistogramConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class MeterFilterConfigPropertiesTest {
+
+    private MeterFilterConfigProperties sut;
+
+    @BeforeEach
+    public void setup(){
+        sut = new MeterFilterConfigProperties();
+        sut.getFilter().put("ALL.enabled","false");
+        sut.getFilter().put("ALL.percentiles","0.3,0.4");
+        sut.getFilter().put("ALL.percentileHistogram","false");
+        sut.getFilter().put("ALL.maximumExpectedValue","PT7S");
+        sut.getFilter().put("ALL.minimumExpectedValue","10000000");
+        sut.getFilter().put("ALL.histogramBufferLength","100");
+        sut.getFilter().put("ALL.durationExpiry","PT60S");
+        sut.getFilter().put("ALL.sla","PT0.1S,PT1S");
+        sut.getFilter().put("jvm.enabled","true");
+        sut.getFilter().put("jvm.my.daemon.enabled","false");
+        sut.getFilter().put("jvm.memory.used","enabled");
```

Comment:
Implementation details of interest:

* I didn't support wildcards, since I discussed with some team mates and we couldn't find a very good usecase
* The only special value is `ALL`, so you have a 'root' option that applies to all metrics
* Keys are case insensitive
* Made it so `jvm.memory.used.enabled: true` and `jvm.memory.used: enabled` both work since the second one seems to be a common usecase (and read much cleaner when flattened with many other config values)

Reply:
I think wildcard will be useful to turn off or configure metrics emanating from a system that has a common prefix (e.g. `hikari.foo`, `hikari.bar`)

Reply:
Boy, you're right. The PTxS syntax looks ugly in the context of SLAs.

Reply:
Hierarchical config is supported `hikari.enabled: false` works, `hikari.*.thing.enabled: false` won't.

I'm very tempted to code a simple parser that only supports a number and a unit. (100ms, 10s).